Competition2017/src/competition/subsystems/BaseXCANTalonSpeedControlledSubsystem.java

```diff
@@ -151,12 +164,13 @@ public void updatePeriodicData() {
         systemTalonError.set(masterMotor.getClosedLoopError());
         
         if(enablesystemLogging.get()){
-            double currentTime = Timer.getFPGATimestamp();
-            // Format: time, voltage, error, speed
-            log.info(currentTime + "," 
-                 + systemOutputPower.get() + "," 
-                 + systemTalonError.get() + ","
-                 + systemCurrentSpeed.get());
+            if(!telemetryLogger.isInitialized()) {
+                telemetryLogger.initialize(name, systemOutputPower, systemTalonError, systemCurrentSpeed);
+            }
+            telemetryLogger.collectNewDataset();
+        }
+        else if(telemetryLogger.isInitialized()) {
+            telemetryLogger.uninitialize();
```

Comment:
I've looked at this for quite a while and can't see anything wrong. The only thing that would make sense is if you are hitting some sort of exception when creating the `out` object, which would cause you to keep initializing over and over.

But then, you wouldn't see the log call of "uninitialize", so that doesn't seem reasonable.